pkg/controller/config/process/excluder.go

```diff
@@ -79,8 +81,13 @@ func (s *Excluder) Equals(new *Excluder) bool {
 	return reflect.DeepEqual(s.excludedNamespaces, new.excludedNamespaces)
 }
 
-func (s *Excluder) IsNamespaceExcluded(process Process, namespace string) bool {
+func (s *Excluder) IsNamespaceExcluded(process Process, gvk schema.GroupVersionKind, name, namespace string) bool {
```

Comment:
Do we want to just accept a `runtime.Object` and let the excluder do the namespace unpacking? Or maybe accept a [metav1.Object](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L33-L66)?

We could use similar library functions as used here:

https://github.com/open-policy-agent/gatekeeper/blob/e47d95b3155767785c45d6e723a00f25790b990c/pkg/mutation/mutator.go#L52-L65

By passing all the metadata we can tweak how things work without needing to adjust how the consumers of the namespace excluder behave (e.g. no function signature changes).

This will also provide a more strongly-typed function signature.

Reply:
Updated. However, for webhook `admissionRequest` things get complicated as trying to access `meta.Accessor` ends up with `object does not implement the Object interfaces` error. Workaround I found is to convert `rawExtension` to `runtime.Object` then to `unstructured` to be able to access it as `runtime.Object` and have `meta.Accessor` to be accessible.